### PR TITLE
fix: avoid feeding paths into applicationHash calculation

### PR DESCRIPTION
**Pull request by fiam**
Store ResolveApplicationConfig in a field in CodeGenerationConfig
rather than extending it. This allows us to safely hash ResolveApplicationConfig
without including the paths in CodeGenerationConfig.

### CODE REVIEW

File: packages/golang-client/src/index.ts [hunk 1]
```diff
@@ -7,6 +7,7 @@ import Handlebars from 'handlebars';
 import { clientTemplate } from './client-template';
 import { OperationType } from '@wundergraph/protobuf';
 import Logger from '@wundergraph/sdk/internal/logger';
+import { CodeGenerationConfig } from '@wundergraph/sdk/dist/configure';
```
Comment (StarpTech) [inline packages/golang-client/src/index.ts:10]: Why point to dist? if is not exported, please do it.

File: packages/sdk/src/configure/index.ts [hunk 2]
```diff
@@ -259,7 +259,8 @@ export interface ResolvedWunderGraphConfig {
 	serverOptions?: ResolvedServerOptions;
 }
 
-export interface CodeGenerationConfig extends ResolvedWunderGraphConfig {
```
Comment (StarpTech) [inline packages/sdk/src/configure/index.ts:262]: Could you explain why it makes a difference for hashing?

**Review by StarpTech — CHANGES_REQUESTED** [review]

File: packages/golang-client/src/index.ts (hunk 1 above)
Comment (Pagebakers) [reply packages/golang-client/src/index.ts:10]: Yep /dist was for backwards compat, we should remove it at some point

Comment (fiam) [reply packages/golang-client/src/index.ts:10]: It was not exported. The editor filled the import automatically and I didn't think about it. I'll update it.

File: packages/sdk/src/configure/index.ts (hunk 2 above)
Comment (fiam) [reply packages/sdk/src/configure/index.ts:262]: I've added comments in `CodeGenerationConfig` and `ResolvedWunderGraphConfig`, PTAL

Comment (StarpTech) [reply packages/sdk/src/configure/index.ts:262]: ah now, I got it. Great fix

**Review by StarpTech — APPROVED** [review]
LGTM

**Review by thisisnithin — APPROVED** [review]

**Pull request merged**